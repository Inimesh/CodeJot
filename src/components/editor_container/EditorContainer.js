import React, { useState } from 'react'
import { View, StyleSheet, TextInput, Text } from 'react-native'
import SyntaxHighlighter from 'react-native-syntax-highlighter';
import { docco } from 'react-syntax-highlighter/dist/cjs/styles/hljs';

const EditorContainer = ({ language }) => { // { synHighlight }

  const [code, setCode] = useState("");

  return (
    <View style={styles.editorWindow}>
      <TextInput style={styles.textField}
      placeholder='Type to start coding!'
      onChangeText={newCode => setCode(newCode)}
      autoCapitalize='none'
      multiline={true}
      numberOfLines={10}
      textAlign='left'
      textAlignVertical='top'
      defaultValue={code}
      />
      <SyntaxHighlighter language={language} highlighter="hljs" style={docco}>
        {code}
      </SyntaxHighlighter>
    </View>
  )
}

const styles = StyleSheet.create({
  editorWindow: {
    marginLeft: 12,
    marginRight: 12,
    marginTop: 12,
    marginBottom: 12,
    borderWidth: 1,
    borderRadius: 4,
    // backgroundColor: "red",
  },
  textField: {
    backgroundColor: "#fff",
    paddingLeft: 6,
    paddingRight: 6,
    paddingTop: 6,
    paddingBottom: 6,
    borderRadius: 4,
  }
})

export default EditorContainer