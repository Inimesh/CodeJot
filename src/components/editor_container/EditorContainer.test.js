import React from 'react'
import renderer from 'react-test-renderer'

import EditorContainer from './EditorContainer'

describe('<Display />', () => { 
  it('renders components correctly', () => {
    const tree = renderer.create(<EditorContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

})