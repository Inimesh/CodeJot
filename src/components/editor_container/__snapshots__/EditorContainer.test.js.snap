// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> renders components correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 4,
      "borderWidth": 1,
      "marginBottom": 12,
      "marginLeft": 12,
      "marginRight": 12,
      "marginTop": 12,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    defaultValue=""
    multiline={true}
    numberOfLines={10}
    onChangeText={[Function]}
    placeholder="Type to start coding!"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 4,
        "paddingBottom": 6,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 6,
      }
    }
    textAlign="left"
    textAlignVertical="top"
    underlineColorAndroid="transparent"
  />
  <RCTScrollView
    horizontal={true}
    style={
      Object {
        "backgroundColor": "#f8f8ff",
        "overflow": "scroll",
        "padding": 8,
      }
    }
  >
    <View>
      <RCTScrollView>
        <View>
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "Menlo-Regular",
                "fontSize": 12,
                "height": 17,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "Menlo-Regular",
                  "fontSize": 12,
                  "height": 17,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "Menlo-Regular",
                    "fontSize": 12,
                    "height": 17,
                  }
                }
              >
                
              </Text>
            </Text>
          </Text>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
